x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/fielddata/ScriptBooleanDocValues.java

```diff
@@ -21,19 +23,33 @@
 
     @Override
     public boolean advanceExact(int docId) {
-        script.runForDoc(docId);
-        cursor = 0;
-        return script.trues() > 0 || script.falses() > 0;
+        boolean[] values = script.runForDoc(docId);
+        if (values.length == 0) {
+            return false;
+        }
+        trues = falses = cursor = 0;
+        /*
+         * We need to emit all falses before all trues so instead of
+         * sorting we count them all.
```

Comment:
This is the contract for doc values - that the results be sorted. And `false`s are 0 and `true`s are 1.